carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```diff
@@ -0,0 +1,286 @@
+        if (srcDir.isDirectory()) {
+            if (!dstDir.exists()) {
+                if (!dstDir.mkdirs()) {
+                    throw new IOException("Failed to delete " + dstDir.getAbsolutePath());
+                }
+            }
+
+            String[] children = srcDir.list();
+            if (children != null) {
+                for (String child : children) {
+                    copyDirectory(new File(srcDir, child),
+                            new File(dstDir, child));
+                }
+            }
+        } else {
+            copy(srcDir, dstDir);
+        }
+    }
+
+    /**
+     * Copies src file to dst file. If the dst file does not exist, it is created.
+     *
+     * @param src source file
+     * @param dst destination file
+     * @throws IOException
+     */
+    public static void copy(File src, File dst) throws IOException {
+        if (dst.getParentFile() != null && !dst.getParentFile().exists()) {
+            if (!dst.getParentFile().mkdirs()) {
+                throw new IOException("Failed to create " + dst.getAbsolutePath());
+            }
+        }
+
+        InputStream in = new FileInputStream(src);
+        OutputStream out = new FileOutputStream(dst);
+        // Transfer bytes from in to out
+        byte[] buf = new byte[1024];
+        int len;
+        while ((len = in.read(buf)) > 0) {
+            out.write(buf, 0, len);
+        }
+//        } catch (IOException e) {
+//            e.printStackTrace();
+//        }
+    }
+
+    /**
+     * Unzip a given archive to a given destination
+     *
+     * @param archiveFile archive file to be unzipped
+     * @param destination location to put the unzipped file
+     * @throws IOException
+     */
+    public static void unzip(File archiveFile, File destination) throws IOException {
+
+        try(FileInputStream fis = new FileInputStream(archiveFile);
+            ZipInputStream zis = new
+                    ZipInputStream(new BufferedInputStream(fis))) {
+            ZipEntry entry;
+
+            while ((entry = zis.getNextEntry()) != null) {
+                int count;
+                byte data[] = new byte[BUFFER];
+                File file = new File(destination, entry.getName());
+                if (entry.getName().endsWith("/")) {
+                    if (!file.exists() && !file.mkdirs()) {
+                        throw new IOException("Failed to create directories at " + file.getAbsolutePath());
+                    }
+                    continue;
+                }
+                if (file.getParentFile() != null && !file.getParentFile().exists()) {
+                    if (!file.getParentFile().exists() && !file.getParentFile().mkdirs()) {
+                        throw new IOException("Failed to create directories at " + file.getAbsolutePath());
+                    }
+                }
+                try(FileOutputStream fos = new FileOutputStream(file);
```

Comment:
This is the migrated code. It is refactored in https://github.com/wso2/carbon-maven-plugins/pull/8